### DISCUSSION TITLE
Request for Documentation/Explanation on Azure Resource Use

### DISCUSSION THREAD

**Discussion by acambitsis** [Azure-Samples/graphrag-accelerator#103] [Q&A] [answered]
Hello everyone,

I am seeking some guidance regarding the usage and purpose of the following resources in the project:

1. **Azure AI Search Resource**: Could someone please explain why this resource is deployed and what it is used for?

2. **CosmosDB**: I need clarification on what this database is used for within the setup.

For context, I have run the Python library locally and have a basic understanding of its functionality. 

Any insights or documentation related to the above would be greatly appreciated.

Thank you in advance!

Best regards,  
Andreas

**Comment by jasonclane** [comment]
I was about to post something similar, but you beat me to it. :)

I spent a few days dealing with deployment failures due to Azure quota limits (I was testing on a Pay as you go account that I had converted from a free trial).

It sure would be good to have some documentation around how many resources of what types were needed in addition to the specific description/explanation of how the resource is used. I had to move regions even since the region I started with had locked quotas on certain types of compute resources.

**Comment by timothymeyers** [comment, marked as answer]
Hi @acambitsis and @jasonclane - Thanks for the question. We'll see about adding additional documentation explaining the resources.

re: your question,

1. **Azure AI Search** - This is used for storing vector embeddings on the dataset. During indexing, vector embeddings are computed and stored in Azure AI Search, and during queries, a vector similarity search (think: standard RAG) is performed in conjunction with using the GraphRAG knowledge graph as part of the query algorithm. The [graphrag python library](https://github.com/microsoft/graphrag/) defaults to using a local LanceDB vectorstore for this purpose, but in a cloud deployment, the library natively supports using Azure AI Search as the cloud version. I think the graphrag library actually has a few PRs in its queue to support other vector stores natively as well.
2. **Cosmos DB** - CosmosDB is primarily used to store some configuration information (e.g., data store and index name lookups), job status information, and knowledge graph metadata for quick retrieval (e.g. entity or node information). I'm not sure if there's a local equivalent in the graphrag library since most of this is just to help with tracking runs 'in the cloud.'

**Reply by acambitsis** [reply]
Thanks very much for the explanation. I was wondering, given your robust setup with Kubernetes, private networking, and various Azure services, are you positioning GraphRAG to potentially become an Azure managed service, or is this architecture primarily for internal/client-specific deployments?"
To provide some context, the current deployment configuration can consume more cost than is warranted for our context.  So we need to decide if it's worthwhile tailoring it to our circumstances or waiting for a managed service.
Thanks.

**Reply by timothymeyers** [reply]
The intent for this architecture sample is to show how `graphrag` could be deployed in Azure. While it should not be used as-is for a production system, it lays the groundwork for and shows how to set up for security and scale.

I don't really have any information about a future managed service, but I hope you'll take this example and run with it.
